### PR TITLE
feat(event): add TreeAttachedPost

### LINKED ISSUE #1869: Event.TreeAttachedPost
Event fired following `on_attach` call.

Context:
#1858 
https://github.com/mrjones2014/legendary.nvim/issues/289#issuecomment-1364618688

[Legendary](https://github.com/mrjones2014/legendary.nvim) has a requirement to access / mutate user's mappings set via their `on_attach`.

### PR DESCRIPTION
**Pull request by alex-courtis**
closes #1869

### CODE REVIEW

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -1639,6 +1639,11 @@ e.g. handler for node renamed: >
                 handler parameters: ~
                   {folder_name}  `{string}` Absolute path to the removed folder.
 
+- Event.OnAttachPost
```
Comment (gegoune) [inline doc/nvim-tree-lua.txt:1642]: Do we need `On` prefix here? Seems like other events don't have those. `AttachPost` sounds bit off though. How about `TreeAttached`? Might want to make it `TreeAttachedPost` if there is a chance of `TreeAttachedPre` being added as well.

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:1642]: That makes sense. Telescope uses attach and gitsigns only adds `on_` for the callback itself.

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:1642]: If we ever handle multiple trees/tabs, that could be added.

Comment (gegoune) [reply doc/nvim-tree-lua.txt]: Thanks, one more option I thought of: `TreeCreated` - maybe it's bit more descriptive than `Attached`. Up to you.

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt]: on_attach happens in the middle of things.

We might want a TreeCreated event before or a TreeReady event after if a need arises :shrug:

Comment (gegoune) [reply doc/nvim-tree-lua.txt]: That makes sense then. Thanks once again.

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt]: Thank you for the diligence.

I'm just super-paranoid about not walling us into any corners with API or Events.

Whilst I have you... could I please ask for a sanity check / test for #1872

Off until next weekend :wave:

**Review by gegoune — APPROVED** [review]

**Pull request merged**